deepchecks/tabular/checks/model_evaluation/weak_segments_performance.py

```diff
@@ -0,0 +1,328 @@
+        Minimum size ratio for segments. Will only search for segments of
+        size >= segment_minimum_size_ratio * data_size.
+    alternative_scorer : Tuple[str, Union[str, Callable]] , default: None
+        Score to show, either function or sklearn scorer name.
+        If None, a default scorer (per the model type) will be used.
+    loss_per_sample: Union[np.array, pd.Series, None], default: None
+        Loss per sample used to detect relevant weak segments. If None, the check calculates loss per sample by using
+         for classification and mean square error for regression.
+    n_samples : int , default: 10_000
+        number of samples to use for this check.
+    n_to_show : int , default: 3
+        number of segments with the weakest performance to show.
+    categorical_aggregation_threshold : float , default: 0.05
+        Categories with appearance ratio below threshold will be merged into "Other" category.
+    random_state : int, default: 42
+        random seed for all check internals.
+    """
+
+    def __init__(
+            self,
+            columns: Union[Hashable, List[Hashable], None] = None,
+            ignore_columns: Union[Hashable, List[Hashable], None] = None,
+            n_top_features: int = 5,
+            segment_minimum_size_ratio: float = 0.05,
+            alternative_scorer: Dict[str, Callable] = None,
+            loss_per_sample: Union[np.array, pd.Series, None] = None,
+            classes_index_order: Union[np.array, pd.Series, None] = None,
+            n_samples: int = 10_000,
+            categorical_aggregation_threshold: float = 0.05,
+            n_to_show: int = 3,
+            random_state: int = 42,
+            **kwargs
+    ):
+        super().__init__(**kwargs)
+        self.columns = columns
+        self.ignore_columns = ignore_columns
+        self.n_top_features = n_top_features
+        self.segment_minimum_size_ratio = segment_minimum_size_ratio
+        self.n_samples = n_samples
+        self.n_to_show = n_to_show
+        self.random_state = random_state
+        self.loss_per_sample = loss_per_sample
+        self.classes_index_order = classes_index_order
+        self.user_scorer = alternative_scorer if alternative_scorer else None
+        self.categorical_aggregation_threshold = categorical_aggregation_threshold
+
+    def run_logic(self, context: Context, dataset_kind) -> CheckResult:
+        """Run check."""
+        dataset = context.get_data_by_kind(dataset_kind)
+        dataset.assert_features()
+        dataset = dataset.sample(self.n_samples, random_state=self.random_state, drop_na_label=True)
+        predictions = context.model.predict(dataset.features_columns)
+        y_proba = context.model.predict_proba(dataset.features_columns) if \
+            context.task_type in [TaskType.MULTICLASS, TaskType.BINARY] else None
+
+        if self.loss_per_sample is not None:
+            loss_per_sample = self.loss_per_sample[list(dataset.data.index)]
```

Comment:
I think you should validate that the user gave you the loss-per-sample correctly